frontend/lib/ui/ProgressBar/ProgressBar.tsx

```diff
@@ -2,28 +2,88 @@ import { cn } from "@kiesraad/util";
 
 import cls from "./ProgressBar.module.css";
 
+export type ProgressBarColorClass =
+  | "default"
+  | "errors_and_warnings"
+  | "unfinished"
+  | "in_progress"
+  | "first_entry_finished"
+  | "definitive"
+  | "not_started";
+
+export type RadiusClass = "no-radius" | "default-radius" | "radius-right" | "radius-left";
+export type PercentageAndColorClass = {
+  percentage: number;
+  class: ProgressBarColorClass;
+};
+
 export interface ProgressBarProps {
-  percent: number;
   id: string;
+  percentagesAndColorClasses: PercentageAndColorClass[];
   title?: string;
   spacing?: "small" | "large";
+  showPercentage?: boolean;
 }
-export function ProgressBar({ percent, id, title, spacing }: ProgressBarProps) {
+export function ProgressBar({
+  id,
+  percentagesAndColorClasses,
+  title,
+  spacing,
+  showPercentage = false,
+}: ProgressBarProps) {
+  const filteredBars = filterOutZeroPercentageBars(percentagesAndColorClasses);
+  const totalBars = filteredBars.length;
+  function filterOutZeroPercentageBars(bars: PercentageAndColorClass[]): PercentageAndColorClass[] {
+    return bars.filter((bar) => bar.percentage > 0);
+  }
+
+  function getRadiusClassForBarIndex(barIndex: number): RadiusClass {
```

Comment:
couldn't you achieve this css-only with :first-of-type and :last-of-type ?

Reply:
nope cause that does not work with classes :cry: so the outer div would be the first to be selected

Reply:
Ended up getting rid of the code in c3454c192dbc8c694a1c85fce5b8ed5ee45d7605 by adding `overflow: hidden` on the outer div as Oliver suggested